_posts/2020-10-19-micrometer-metrics.adoc

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: Micrometer is recommended for metrics with Quarkus 1.9
+date: 2020-10-19
+tags: metrics announcement micrometer
+synopsis: Micrometer is the recommended approach for metrics with Quarkus.
+author: kenfinnigan
+---
+
+Though Quarkus offers extensions for MicroProfile,
+including metrics,
+we're now recommending the use of the Micrometer extension for metrics in Quarkus with the release of 1.9.
+
+== Brief MicroProfile Metrics History
+
+During the creation of the MicroProfile Metrics specification in 2017,
+the most popular metrics library,
+at that time,
+was DropWizard Metrics.
+It was a natural fit for MicroProfile to use DropWizard Metrics as inspiration for a metrics specification.
+
+Since then,
+there have been many advancements with modern monitoring solutions,
+and which is the preferred metrics library in the Java ecosystem.
+
+[[why-micrometer]]
+== Why Micrometer?
+
+During the last 3-4 years,
+there's been a growing focus on the needs of Site Reliability Engineers (SREs) in monitoring the operations of many varied services.
+This focus leads to driving simplicity and ease of use to the SRE world.
+We don't want make the life of an SRE harder!
+
+The design and development of Micrometer focused on the needs of SREs from the very beginning.
+A key aspect to that approach is separating out the metric implementation into the registries,
+enabling different registries to customize how the metric export,
+aligning with a specific monitoring tool.
+Micrometer offers registries for Prometheus, DataDog, New Relic, Stackdriver, and many more.
+This enables the same application to switch the monitoring tool with a single dependency,
+and without adjusting application code utilizing metrics.
+
```

Comment:
Micrometer was built to adapt the changing cloud native landscape while presenting a steady API for applications and libraries to use. Each MeterRegistry implementation encapsulates its specific aggregation and reporting requirements. There are MeterRegistry implementations available for cloud monitoring systems like Prometheus and Statsd, as well as service providers like Datadog, New Relic or Stackdriver. You can more than one of these registries at the same time, too.

While Micrometer has an API for applications to use, practical usage is more focused on the MeterBinder abstraction. MeterBinders allow instrumentation for application libraries to be added later. It is even better when your application framework can register discovered MeterBinders on your behalf.